src/Adapter/Carrier/CarrierThumbnailLogoProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Carrier;
+
+use HelperList;
+use ImageManager;
+use PrestaShop\PrestaShop\Core\Image\ImageProviderInterface;
+use PrestaShop\PrestaShop\Core\Image\Parser\ImageTagSourceParserInterface;
+
+/**
+ * Provides path for Carrier logo in carriers grid
+ */
+class CarrierThumbnailLogoProvider implements ImageProviderInterface
+{
+    /**
+     * @var ImageTagSourceParserInterface
+     */
+    private $imageTagSourceParser;
+
+    /**
+     * @param ImageTagSourceParserInterface $parser
+     */
+    public function __construct(ImageTagSourceParserInterface $parser)
+    {
+        $this->imageTagSourceParser = $parser;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPath($carrierId)
```

Comment:
```suggestion
    public function getPath(int $carrierId): int
```
You probably need to update the interface too

Reply:
Modifying the interface means a BC Break though, I'm not sure we change that for now.

Reply:
Yes, this is really out-of-scope of this PR.